### PR TITLE
[eas-cli] use generated graphql types

### PR DESCRIPTION
**Pull request by jkhales**
# Why

Using generated graphql code types for commands leaves less room for error.

# How

Used pregenerated graphql types where appropriate.

# Test Plan

tested commands still work in a test project.

### CODE REVIEW

File: packages/eas-cli/src/commands/branch/list.ts [hunk 1]
```diff
@@ -37,68 +44,74 @@ export default class BranchList extends Command {
...
-      )
-      .toPromise();
-
-    if (error) {
-      if (error.networkError) {
-        throw new CLIError(`Fetching branches failed: ${error.networkError.message}`);
-      } else {
-        throw new CLIError(error.graphQLErrors.map(e => e.message).join('\n'));
-      }
-    }
```
Comment (jkhales) [inline packages/eas-cli/src/commands/branch/list.ts:95]: replaced this custom error handling with `withErrorHandlingAsync` above.

File: packages/eas-cli/src/commands/branch/view.ts [hunk 2]
```diff
@@ -19,34 +27,16 @@ type TruncatedUpdate = Pick<Update, 'group' | 'message' | 'createdAt' | 'actor'>
...
-            };
-          };
-        },
-        {
-          appId: string;
-          name: string;
-          limit: number;
-        }
-      >(
+      .mutation<ViewBranchQuery, ViewBranchQueryVariables>(
```
Comment (fson) [inline packages/eas-cli/src/commands/branch/view.ts:39]:
```suggestion
      .query<ViewBranchQuery, ViewBranchQueryVariables>(
```

File: packages/eas-cli/src/commands/build/cancel.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ import { confirmAsync, selectAsync } from '../../prompts';
 async function cancelBuildAsync(buildId: string): Promise<Pick<Build, 'id' | 'status'>> {
   const data = await withErrorHandlingAsync(
     graphqlClient
-      .mutation<{ build: { cancel: { id: string; status: BuildStatus } } }>(
+      .mutation<CancelBuildMutation>(
```
Comment (fson) [inline packages/eas-cli/src/commands/build/cancel.ts:22]:
```suggestion
      .mutation<CancelBuildMutation, CancelBuildMutationVariables>(
```

File: packages/eas-cli/src/commands/build/cancel.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ import ora from 'ora';
 import { platformEmojis } from '../../build/constants';
 import { Platform } from '../../build/types';
 import { graphqlClient, withErrorHandlingAsync } from '../../graphql/client';
-import { AppPlatform, Build, BuildStatus } from '../../graphql/generated';
+import { AppPlatform, Build, BuildStatus, CancelBuildMutation } from '../../graphql/generated';
```
Comment (fson) [inline packages/eas-cli/src/commands/build/cancel.ts:9]:
```suggestion
import { AppPlatform, Build, BuildStatus, CancelBuildMutation, CancelBuildMutationVariables } from '../../graphql/generated';
```

**Review by fson — CHANGES_REQUESTED** [review]
Thanks, this makes the types much more useful.

**Review by fson — APPROVED** [review]

**Pull request merged**